rust/lance-graph/tests/test_datafusion_pipeline.rs

```diff
@@ -3123,6 +3189,246 @@ async fn test_avg_without_alias_has_descriptive_name() {
+        .unwrap();
+
+    assert_eq!(result.num_rows(), 1);
+
+    // Column should be named "min(p.age)"
+    let col = result.column_by_name("min(p.age)");
+    assert!(
+        col.is_some(),
+        "Expected column named 'min(p.age)' but schema is: {:?}",
+        result.schema()
+    );
+}
+
+#[tokio::test]
+async fn test_min_with_grouping() {
+    let mut person_batch = create_person_dataset();
+    person_batch = add_new_person_to_dataset(&person_batch, 6, "Cindy", 10, Some("New York"));
+    let config = GraphConfig::builder()
+        .with_node_label("Person", "id")
+        .build()
+        .unwrap();
+
+    // One person per city in this dataset (including NULL), so min(age) == that person's age
+    let query =
+        CypherQuery::new("MATCH (p:Person) RETURN p.city, min(p.age) AS min_age ORDER BY p.city")
+            .unwrap()
+            .with_config(config);
+
+    let mut datasets = HashMap::new();
+    datasets.insert("Person".to_string(), person_batch);
+
+    let result = query
+        .execute(datasets, Some(ExecutionStrategy::DataFusion))
+        .await
+        .unwrap();
+
+    assert_eq!(result.num_rows(), 5);
+
+    let city_col = result
+        .column_by_name("p.city")
+        .unwrap()
+        .as_any()
+        .downcast_ref::<StringArray>()
+        .unwrap();
+
+    let min_col = result
+        .column_by_name("min_age")
+        .unwrap()
+        .as_any()
+        .downcast_ref::<Int64Array>()
+        .unwrap();
+
+    // ORDER BY p.city, NULL comes first per your other tests
+    assert!(city_col.is_null(0)); // David city NULL
+    assert_eq!(min_col.value(0), 40);
+
+    assert_eq!(city_col.value(1), "Chicago"); // Charlie
+    assert_eq!(min_col.value(1), 30);
+
+    assert_eq!(city_col.value(2), "New York"); // Alice
+    assert_eq!(min_col.value(2), 10);
+
+    assert_eq!(city_col.value(3), "San Francisco"); // Bob
+    assert_eq!(min_col.value(3), 35);
+
+    assert_eq!(city_col.value(4), "Seattle"); // Eve
+    assert_eq!(min_col.value(4), 28);
+}
+
+#[tokio::test]
+async fn test_max_property() {
+    let person_batch = create_person_dataset();
+    let config = GraphConfig::builder()
+        .with_node_label("Person", "id")
+        .build()
+        .unwrap();
+
+    let query = CypherQuery::new("MATCH (p:Person) RETURN max(p.age) AS max_age")
+        .unwrap()
+        .with_config(config);
+
+    let mut datasets = HashMap::new();
+    datasets.insert("Person".to_string(), person_batch);
+
+    let result = query
+        .execute(datasets, Some(ExecutionStrategy::DataFusion))
+        .await
+        .unwrap();
+
+    assert_eq!(result.num_rows(), 1);
+
+    let max_col = result
+        .column_by_name("max_age")
+        .unwrap()
+        .as_any()
+        .downcast_ref::<Int64Array>()
+        .unwrap();
+
+    // Ages: 25, 35, 30, 40, 28 => max = 40
+    assert_eq!(max_col.value(0), 40);
+}
+
+#[tokio::test]
+async fn test_max_without_alias_has_descriptive_name() {
```

Comment:
Generally, `MIN` and `MAX` are sharing very similar features, so I think we don't need this many new tests. How about just keeping

- test_min_property
- test_max_property
- test_min_max_with_grouping (testing min and max with grouping in one method)

We can safely remove other MIN/MAX tests.